Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.cpp

```diff
@@ -119,6 +119,60 @@ namespace AzToolsFramework
                 return true;
             }
 
+            bool StoreInstanceInPrefabDom(
```

Comment:
There's a lot of overlap with the other `StoreInstanceInPrefabDom`, it might be worth narrowing to one implementation that takes `referencedAssets` as an optional pointer to avoid implementation drift (this already is the only implementation that adds `AZ::Data::SerializedAssetTracker` to the metadata, which I assume is probably correct but would struggle to audit at a glance).

Reply:
Moved the shared logic into an internal function.